include/odp/api/spec/ipsec.h

```diff
@@ -71,6 +71,48 @@ typedef enum odp_ipsec_op_mode_t {
 
 } odp_ipsec_op_mode_t;
 
+/**
+ * IPSEC TEST SA parameter type
+ */
+typedef enum odp_ipsec_test_sa_param_type_t {
+	/** Next sequence number value */
+	ODP_IPSEC_TEST_SA_SEQ_NUM = 0,
+
+	/** Highest authenticated sequence number */
+	ODP_IPSEC_TEST_SA_ANTIREPLAY_WINDOW_TOP
+
+} odp_ipsec_test_sa_param_type_t;
+
+/**
+ * IPSEC TEST SA parameter value
+ */
+typedef union odp_ipsec_test_sa_param_val_t {
+	/** Next sequence number value */
+	uint64_t seq_num;
+
+	/** Highest authenticated sequence number */
+	uint64_t antireplay_window_top;
+
+} odp_ipsec_test_sa_param_val_t;
```

Comment:
I wonder if the API would be more flexible if the above fields were both of the same "generic integer" type. This way most parameters would not need their own fields in this union. OTOH, once someone adds something like and IV or bytes to be used as random data or something like that, we would need an array of bytes. so new parameters may required new values in the union even if we combine the simple types now.

Reply:
> I wonder if the API would be more flexible if the above fields were both of the same "generic integer" type.

Did you mean something like:
union {
  uint64_t u64;
  uint8_t buffer[MAX_SIZE];
}

Reply:
After discussing with Petri, I think it is better to leave this as proposed, i.e. there is a separate union field for each possible parameter.

Reply:
Ack.